compiler/rustc_resolve/src/late.rs

```diff
@@ -1343,6 +1343,32 @@ impl<'a: 'ast, 'b, 'ast, 'tcx> LateResolutionVisitor<'a, 'b, 'ast, 'tcx> {
         }
     }
 
+    fn resolve_defines_attr(&mut self, attrs: &[Attribute]) {
+        for attr in attrs {
+            // for `#[defines(Path, other::Path)]` attributes we need to resolve the paths
+            if attr.has_name(sym::defines) {
+                for item in attr.meta_item_list().unwrap() {
+                    let item = item.meta_item().unwrap();
+                    match self.resolve_path(
+                        &Segment::from_path(&item.path),
+                        Some(Namespace::TypeNS),
+                        None,
+                    ) {
+                        // TODO: error reporting and tests that go through these code paths
+                        PathResult::Module(_) => todo!(),
+                        PathResult::NonModule(partial_res) => {
+                            let id = self.r.next_node_id();
```

Comment:
Are these ids necessary?
Probably simpler to push the resolutions straight to `self.r.defines.entry(attr.id)`, without doing `record_partial_res`.